src/rules/base.jl

```diff
@@ -37,7 +37,7 @@
 @scalar_rule(conj(x), Wirtinger(Zero(), One()))
 @scalar_rule(adjoint(x), Wirtinger(Zero(), One()))
 @scalar_rule(transpose(x), One())
-@scalar_rule(abs(x), sign(x))
+@scalar_rule(abs(x), x isa Real ? sign(x) : Wirtinger(x' / 2Ω, x / 2Ω))
```

Comment:
```suggestion
@scalar_rule(abs(x::Real), sign(x))
@scalar_rule(abs(x::Complex), Wirtinger(x' / 2Ω, x / 2Ω))
```

Reply:
Now that we have that, this is definitely nicer!